templates/components/engines/typescript/agent/tools/interpreter.ts

```diff
@@ -0,0 +1,171 @@
+import { CodeInterpreter, Logs, Result } from "@e2b/code-interpreter";
+import type { JSONSchemaType } from "ajv";
+import fs from "fs";
+import { BaseTool, ToolMetadata } from "llamaindex";
+import crypto from "node:crypto";
+import path from "node:path";
+
+export type InterpreterParameter = {
+  code: string;
+};
+
+export type InterpreterToolParams = {
+  metadata?: ToolMetadata<JSONSchemaType<InterpreterParameter>>;
+  apiKey?: string;
+};
+
+export type InterpreterToolOuput = {
+  isError: boolean;
+  logs: Logs;
+  extraResult: InterpreterExtraResult[];
+};
+
+type InterpreterExtraType =
+  | "html"
+  | "markdown"
+  | "svg"
+  | "png"
+  | "jpeg"
+  | "pdf"
+  | "latex"
+  | "json"
+  | "javascript";
+
+export type InterpreterExtraResult = {
+  type: InterpreterExtraType;
+  // url: string;
+  filename: string;
+};
+
+const DEFAULT_META_DATA: ToolMetadata<JSONSchemaType<InterpreterParameter>> = {
+  name: "interpreter",
+  description: `
+  - You are a Python interpreter.
+  - You are given tasks to complete and you run python code to solve them.
+  - The python code runs in a Jupyter notebook. Every time you call \`interpreter\` tool, the python code is executed in a separate cell. It's okay to make multiple calls to \`interpreter\`.
+  - Display visualizations using matplotlib or any other visualization library directly in the notebook. Shouldn't save the visualizations to a file, just return the base64 encoded data.
+  - You can install any pip package (if it exists) if you need to but the usual packages for data analysis are already preinstalled.
+  - You can run any python code you want in a secure environment.
```

Comment:
This seems more like a system prompt than a function description.

Reply:
i agree, this is the tool description that is used by the LLM to call the tool, should be something simple like:
`Execute python code in a Jupyter notebook cell and returns any result, stdout, stderr, display_data, and error.`
The rest should be in the system prompt of the agent: I guess this means that we need a different system prompt in the .env file if we select the code interpreter tool

for the difference between the system prompt and the tool description, see prompting at https://e2b.dev/docs/hello-world/js

Reply:
I think we don't need the tool system prompt during run-time, just at code generation time.
So how about: 
If the user selects a specific tool, we suggest to use a specific system prompt.

Means you can add the tool system prompt to helpers/tools.ts
and when you generate the env variable you use the system prompt of the selected tool (for now you can just concat if the user selects more than one tool with a system prompt)